src/time/Waveform.hpp

```diff
@@ -48,6 +52,9 @@ class Waveform {
   /// Data values of time windows.
   Eigen::MatrixXd _timeWindows;
 
+  /// number of valid samples in _timeWindows
+  int _numberOfValidSamples;
```

Comment:
I guess we could find a better name here. The variable is only needed for extrapolation, right? Or do you see a use case for real waveform stuff as well? If not, we could use sth like `maxPossibleExtrapolationOrder`?

Reply:
This is a duplicate